packages/contractkit/src/wallets/aws-hsm-wallet.ts

```diff
@@ -29,20 +36,23 @@ export default class AwsHsmWallet extends RemoteWallet implements Wallet {
     const { Keys } = await this.kms.listKeys().promise()
     const addressToSigner = new Map<Address, Signer>()
     for (const { KeyId } of Keys!) {
+      if (!KeyId) {
+        throw new Error(`Missing KeyId in KMS response object ${Keys!}`)
+      }
       try {
-        const { KeyMetadata } = await this.kms.describeKey({ KeyId: KeyId! }).promise()
+        const { KeyMetadata } = await this.kms.describeKey({ KeyId }).promise()
         if (!KeyMetadata?.Enabled) {
           continue
         }
 
-        const publicKey = await this.getPublicKeyFromKeyId(KeyId!)
-        addressToSigner.set(
-          publicKeyToAddress(bigNumberToBuffer(publicKey, 64).toString('hex')),
-          new AwsHsmSigner(this.kms, KeyId!, publicKey)
-        )
+        const publicKey = await this.getPublicKeyFromKeyId(KeyId)
+        const address = getAddressFromPublicKey(publicKey)
+        addressToSigner.set(address, new AwsHsmSigner(this.kms, KeyId, publicKey))
       } catch (e) {
-        // todo: what does the error look like here
-        throw e
+        // Safely ignore non-secp256k1 keys
```

Comment:
Yes, good idea on logging. We don't want to throw the exception if it fails because of non-secp256k1 keys. If you try to call `getPublicKey` on a key that doesn't support it, then it will throw the `UnsupportedOperationException` exception. We can ignore this type of exception since it won't happen with secp256k1 keys. If it's another type of exception, it's unexpected and we should throw.